### DISCUSSION TITLE
Amateur question. Java.

### DISCUSSION THREAD

**Discussion by wc-yaa** [NateIsStalling/Basic4GLj#46] [General]
Hello! This programming software looks nice! I am quite curious. I have a question. Why make a java version of Basic4GL? What is the benefit of Java?

**Comment by NateIsStalling** [comment]
Hello @wc-yaa! Thanks for checking out my project and good question!

My main reason for making a Java version of Basic4GL is to support Basic4GL on more platforms than just Windows, and Java felt like the best cross-platform option when I started the project since I originally wanted to create an Android version of Basic4GL and Java was the primary language for Android development at the time.

I believe there's quite a few good cross-platform options these days with C++ QT, C# .NET MAUI, Kotlin native, and progressive web app frameworks all being potentially good options, but Java has the benefit of being a framework I'm familiar with and I have been able to find Java libraries to support the majority of Basic4GL functionality on Windows, MacOS, and Linux.

**Reply by wc-yaa** [reply]
Hi....thanks for explaining! I hope that the version 1.0 of Basic4GLj will be finished without any complex problem. Yeah!
